dvc/remote/local.py

```diff
@@ -188,6 +188,8 @@ def hardlink(self, from_info, to_info):
                     src=str(from_info), dest=str(to_info)
                 )
             )
+            # skip verifying that hardlink is created
+            self.cache_type_confirmed = True
```

Comment:
What will happen if we have 2 files that we need to create. First one is an empty one and the second one is a non-empty one and if hardlinks are not actually supported? Will link detection logic be able to recover from it even if it sees `            self.cache_type_confirmed = True` after empty file creation?

Reply:
Looks like it does not. Thanks for pointing that out.

Reply:
Looks like the `_verify_link` only verified the first time and later set `self.cache_type_confirmed = True`, so, no checks seem to have been done afterwards. Same as this current PR behavior which needs to be fixed.

https://github.com/iterative/dvc/blob/05cc023164766fbbec91642d7d8af513c8d4079b/dvc/remote/base.py#L377-L386